server/src/chaos/models.rs

```diff
@@ -1,41 +1,573 @@
 
-    let res = diesel::sql_query(include_str!("query.sql"))
-        .bind::<Date, _>(publication_period.1)
-        .bind::<Date, _>(publication_period.0)
-        .bind::<Date, _>(publication_period.1)
-        .bind::<Array<Text>, _>(contributors)
-        .bind::<Int4, _>(config.chaos_batch_size as i32)
-        .bind::<Int4, _>(config.chaos_batch_size as i32)
-        .load::<ChaosDisruption>(&connection);
-
-    if let Err(ref err) = res {
-        println!("{}", err);
+    let mut disruption_maker = DisruptionMaker::default();
+
+    let mut offset_query = 0;
+
+    loop {
+        info!("Querying chaos database");
+        let res = diesel::sql_query(include_str!("query.sql"))
+            .bind::<Date, _>(publication_period.1)
+            .bind::<Date, _>(publication_period.0)
+            .bind::<Date, _>(publication_period.1)
+            .bind::<Array<Text>, _>(contributors)
+            .bind::<Int4, _>(config.chaos_batch_size as i32)
+            .bind::<Int4, _>(offset_query as i32)
+            .load::<ChaosDisruption>(&connection);
+        // Increment offset in query
+        offset_query += config.chaos_batch_size;
+
+        let rows = res?;
+        if rows.is_empty() {
+            break;
+        }
+
+        info!("Converting database rows into Disruption");
+        for row in rows {
+            if let Err(ref err) = disruption_maker.read_disruption(&row) {
+                error!("{}", err);
+            }
+        }
+    }
+
+    info!("Disruptions ready to be applied");
+
+    Ok(disruption_maker.disruptions.into_values().collect())
+}
+
+// Each ChaosDisruption contains only a part of a Disruption
+// So In order to construct a Disruption we parse & merge each ChaosDisruption
+// this is the main goal of DisruptionMaker
+#[derive(Default)]
+struct DisruptionMaker {
+    // When we receive a ChaosDisruption we can create a new Disruption if the disruption id is new (ie not in disruptions_set)
+    // or we update an already created Disruption stored in disruptions HashMap
+    pub(crate) disruptions: HashMap<Uid, chaos_proto::chaos::Disruption>,
+
+    // For each disruption we can have multiple impact, tag and property
+    // In order to push unique impact, tag and property we use HashSet/HashMap
+    // to check if the corresponding tag, impact, property has already been pushed in a disruption
+    // Note : tags and property do not need to be completed/updated
+    // but an impact can be completed so we use a HashMap<Uid, usize>
+    // in order to locate the correct position of impact in disruption.impacts[] vector
+    pub(crate) tags_set: HashSet<Uid>,
+    pub(crate) properties_set: HashSet<(String, String, String)>, //(type, key, value)
+    pub(crate) impacts_set: HashMap<Uid, usize>,
+
+    pub(crate) impact_object_set: ImpactMaker,
+}
+
+impl DisruptionMaker {
+    pub fn read_disruption(&mut self, row: &ChaosDisruption) -> Result<(), Error> {
```

Comment:
```suggestion
    pub fn read_disruption(&mut self, row: &ChaosRow) -> Result<(), Error> {
```
Maybe rename ChaosDisruption -> ChaosRow, or something else that suggest  this is just a "part" of a disruption